app/store/sagas/athlete.js

```diff
@@ -1,19 +1,31 @@
 import { call, select, put } from "redux-saga/effects";
 
-import { updateEntity } from "../actions/entities";
+import { setEntity, setEntities } from "../actions/entities";
 
 import { token } from "../state/appState/selectors";
 
 import { getAthleteDetails } from "../services/athlete";
 
 export function* getStats(athleteID) {
   const accessToken = yield select(token);
-  const { details, error } = yield call(
+  const { entities, error } = yield call(
     getAthleteDetails,
     accessToken,
     athleteID
   );
-  if (!error && details) {
-    yield put(updateEntity(athleteID, "athletes", details));
+  if (!error && entities) {
+    yield put(setEntity("athletes", entities.athletes));
+  }
+}
+
+export function* getDetails(athleteID) {
+  const accessToken = yield select(token);
+  const { entities, error } = yield call(
+    getAthleteDetails,
+    accessToken,
+    athleteID
+  );
+  if (!error && entities) {
+    yield put(setEntities(entities));
   }
```

Comment:
definitely! This will be done in a near future! error handling process to be done!
👍